src/Sarif/PeekableStream.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.Sarif.Readers
+{
+    /// <summary>
+    /// This stream provides a very limited Seekability for Streams which would benefit from an ersatz Seek(Origin)
+    /// capability, but which only need it at the beginning of a Stream.  Ie: to categorize binary versus textual 
+    /// data.
+    /// </summary>
+    internal class PeekableStream : Stream
+    {
+        private readonly Stream underlyingStream;
+        private byte[] rewindBuffer;
+        private int cursor;
+        public PeekableStream(Stream underlyingStream, int initialPeekWindow)
```

Comment:
NIT: introduce a line before declaring your ctor to separate from the field decls.